### ISSUE TITLE
CommentsRefactoring: Don't nest block comments

### ISSUE THREAD

**Issue by cal101** [JnRouvignac/AutoRefactor#164]
do not change 

```
// bla
// /** x y */
// some-unused-code
```
to

```
   /**
      * bla
      /** x y */
   */
```

because new comment is closed too early

**Comment by cal101** [comment]
The original issue may be fixed with b9306ce5199fb81a067e498f4cd2b0705bb028ea

but the lines

```  
private String addPeriodAtEndOfFirstLine(Javadoc node, String comment) {
...
        final Matcher matcher = JAVADOC_WITHOUT_PUNCTUATION.matcher(beforeFirstTag);
        if (matcher.matches()) {
```

may need excessive CPU time.

The Pattern looks really strange to me. Can you explain the intention of the Pattern and the Expressions used to reach to it in somewhat detail?
The usage of "?:" looks garbled to me. Why does this need more than one "?:" ?

**Comment by cal101** [comment]
Fixed in 8a52501e1772f8b6e40b566a74ba866f2471af76

**Comment by JnRouvignac** [comment]
Fixed? Are you sure? The fix does not seem related at all?

**Comment by cal101** [comment]
Thats the version I tested against. I guess the actual fix was in b9306ce5199fb81a067e498f4cd2b0705bb028ea.
Should have been used more words...

**Comment by JnRouvignac** [comment]
Ok.
For the record, it is a duplicate of #149
